<!DOCTYPE html>
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    <title>Blackbird - Fork of Phinch</title>
    <meta name="description" content="Blackbird is a fork of Phinch, an open-source framework for visualizing biological data. Whether it's genes, proteins, or microbial species, Blackbird provides an interactive visualization tool that allows users to explore and manipulate large biological datasets." />
    <meta name="keywords" content="data viz, data visualization, sloan founcation, uc davis, biom, biological data, open-source, framework" />
	<link rel="stylesheet" href="css/utility.css">
	<link rel="stylesheet" href="css/home.css">

    <!-- Facebook Meta Tags -->
    <meta property="og:title" content="Blackbird - A fork of Phinch, framework for visualizing bio data">
    <meta property="og:image" content="img/blackbird_logo.png">
    <meta property="og:site_name" content="Blackbird">

	<script src="lib/jquery-1.10.1.min.js"></script>
    <!-- Google Analytics -->
	<script>
		var gaProperty = 'UA-83528924-1';

		// Disable tracking if the opt-out cookie exists.
		var disableStr = 'ga-disable-' + gaProperty;
		if (document.cookie.indexOf(disableStr + '=true') > -1) {
			window[disableStr] = true;
			$('document').ready(function () {
				$('#ga-optout').hide();
				$('#ga-optedout').show();
			});
		}

		function openGaDialog() {
			$('#dialog-ga').show();
			$('#dialog-ga').dialog({ modal: true });
            $('.ui-button-text').html('<i class="icon-remove icon-large"></i>');
		}

		// Opt-out function
		function gaOptout() {
			document.cookie = disableStr + '=true; expires=Thu, 31 Dec 2099 23:59:59 UTC; path=/';
			window[disableStr] = true;
			$('#ga-optout').hide();
			$('#ga-optedout').show();
		}

		(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
					(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
				m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
		})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

		ga('create', 'UA-83528924-1', 'auto', {
			anonymizeIp: true
		});
		ga('send', 'pageview');
	</script>
	
	<!-- Disable Console Msg -->
	<script type="text/javascript">
		if (!(window.console && console.log)) {(function() {
			var noop = function() {};
			var methods = ['assert', 'clear', 'count', 'debug', 'dir', 'dirxml', 'error', 'exception', 'group', 'groupCollapsed', 'groupEnd', 'info', 'log', 'markTimeline', 'profile', 'profileEnd', 'markTimeline', 'table', 'time', 'timeEnd', 'timeStamp', 'trace', 'warn'];
			var length = methods.length;
			var console = window.console = {};
			while (length--) { console[methods[length]] = noop;}
		}()); }
	</script>
</head>

<body>
	<section id = "top_sec">
		<section id = "header">
			<p class = "logo" style="position: relative">
				<a href="index.html"><img src="img/blackbird_logo.png" height="50px"></a>
				<b id="blackbird_version">1.2.1</b>
			</p>
			<ul id = "menu">
				<li><a href="#about">ABOUT</a></li>
				<li id = "help">HELP</li>
			</ul>

	        <div id="dialog-modal">
	            <h2>How to Use</h2>
	            <p class="descPara">

	                <span>Instructions</span></br></br>
	                This visualization framework aims to address current bottlenecks in the analysis of large sequence datasets (rRNA amplicons, metagenomes), helping researchers analyze high-throughput datasets more efficiently. Phich takes advantage of standard outputs from computational pipelines in order to bridge the gap between biological software (e.g. QIIME) and existing data visualization capabilities (harnessing the scalability of WebGL and HTML5 in a browser-based tool). </br></br>
	                Phinch currently supports downstream analyses of <b>.biom files</b> ( <a href="http://biom-format.org/" target="_blank">Biological Observation Matrix</a>, a JSON-formatted file type typically used to represent marker gene OTUs or metagenomic data). <b>All sample metadata and taxonomy/ontology information MUST be embedded in the .biom file before being uploaded into Phinch</b>.</br></br>
	                In QIIME (version 1.7 or later), users can prepare the .biom file by executing the following commands:</br></br>
	                First, construct an OTU table:</br></br>
	                     <code>make_otu_table.py -i final_otu_map_mc2.txt -o otu_table_mc2_w_tax.biom -t rep_set_tax_assignments.txt</code></br></br>
	                Where your input file (-i) is your OTU Map (defining clusters of raw sequences reads), and taxonomy file (-t) contains the taxonomy or gene ontology strings that correspond to each OTU.</br></br>
                	Second, add your sample metadata to your .biom file. In QIIME version 1.8 this can be done using the following command:</br></br>
                    	<code>biom add-metadata -i otu_table_mc2_w_tax.biom -o otu_table_mc2_w_tax_and_metadata.biom -m sample_metadata_mapping_file.txt</code></br></br>
                	In QIIME version 1.7 or below, you can add metadata with the following command:</br></br> 
	                    <code>add_metadata.py -i otu_table_mc2_w_tax.biom -o otu_table_mc2_w_tax_and_metadata.biom -m sample_metadata_mapping_file.txt</code></br></br>
	                Where your input file (-i) is your .biom file from the previous step, and your mapping file (-m) is a tab-delimited file containing sample metadata (<a href="http://qiime.org/documentation/file_formats.html#metadata-mapping-files" target="_blank">formatted according to these QIIME instructions</a>).</br></br>
	                After these two steps, you're ready to upload.</br></br>
	                If you want to visualize biological data currently formatted as a tab-delimited text file (e.g. the style of OTU tables produced by older versions of QIIME, the style of OTU tables produced by older versions of QIIME, or any other type of genomic data that can be reprsented in matrix format), <a href="https://github.com/biom-format/biom-format/blob/df81277857a553e7e5c9679924e09861d8a5f61f/doc/documentation/biom_conversion.rst" target="_blank">please refer to this documentation for conversion instructions</a>. Phinch supports both "sparse" and "dense" BIOM formats (although <b>sparse .biom files are highly recommended, since the file size is much smaller</b>). </br></br>

	                <span>Some important notes on metadata</span></br></br>
	                In order to be properly detected, all date/time metadata must be standardized according to MIxS standardized format (<a href="http://wiki.gensc.org/index.php?title=MIxS" target="_blank">more information at the Genomic Standards Consortium wiki</a>), and entered into one column in your original sample metadata mapping file, as follows:</br></br>
	                [YYYY]-[MM]-[DD]T[hh]:[mm]:[ss]-[Z]</br>
	                This date format lists the year, month, and day, followed by a 24hr timestamp with a UTC offset (Z). Inclusion of timestamp and UTC offset are both optional; metadata columns can include date only. For example, metadata for a sample collected at 2:30pm EST on May 4, 2007 would be entered as: 2007-04-05T14:30:00-05:00</br></br>
	                Similarly, any geographic coordinates or GPS data must be entered as <a href="http://en.wikipedia.org/wiki/Decimal_degrees" target="_blank"></a>decimal degrees (the format used by GoogleMaps, e.g. -90.017926).  We recommend using separate columns labeled “Latitude” and “Longitude” in your original sample metadata mapping file, to ensure that GPS metadata is correctly detected.</br></br>

	                To label your samples in Phinch and export graphics with human-readable IDs, include a column in your metadata mapping file with the header labelled as “phinchID” (these entries can be the same or different as the first SampleID column). The phinchID values will be pulled through into the visualizations to populate graph axes. If this column is not included, an arbitrary numerical ID will be assigned to each sample.</br></br>
	                                
	            </p>            
	        </div>

			<div id="dialog-ga" style="display:none; overflow-y:auto">
				<h2>Google Analytics</h2>
				<p>

					<h4>English</h4>
					This website uses Google Analytics, a web analytics service provided by Google, Inc. (“Google”). Google Analytics uses “cookies”, which are text files placed on your computer, to help the website analyze how users use the site. The information generated by the cookie about your use of the website will be transmitted to and stored by Google on servers in the United States .
					In case IP-anonymisation is activated on this website, your IP address will be truncated within the area of Member States of the European Union or other parties to the Agreement on the European Economic Area. Only in exceptional cases the whole IP address will be first transfered to a Google server in the USA and truncated there. The IP-anonymisation is active on this website.
					Google will use this information on behalf of the operator of this website for the purpose of evaluating your use of the website, compiling reports on website activity for website operators and providing them other services relating to website activity and internet usage.
					The IP-address, that your Browser conveys within the scope of Google Analytics, will not be associated with any other data held by Google. You may refuse the use of cookies by selecting the appropriate settings on your browser, however please note that if you do this you may not be able to use the full functionality of this website. You can also opt-out from being tracked by Google Analytics with effect for the future by downloading and installing Google Analytics Opt-out Browser Addon for your current web browser: <a href="http://tools.google.com/dlpage/gaoptout?hl=en">http://tools.google.com/dlpage/gaoptout?hl=en</a>.
					As an alternative to the browser Addon or within browsers on mobile devices, you can <a href="javascript:gaOptout()">click this link</a> in order to opt-out from being tracked by Google Analytics within this website in the future (the opt-out applies only for the browser in which you set it and within this domain). - See more at: http://rechtsanwalt-schwenke.de/google-analytics-rechtssicher-nutzen-anleitung-fuer-webmaster/#sthash.jLLgpAZg.dpuf. An opt-out cookie will be stored on your device, which means that you'll have to click this link again, if you delete your cookies.

					<h4>German</h4>
					Diese Website benutzt Google Analytics, einen Webanalysedienst der Google Inc. („Google“). Google Analytics verwendet sog. „Cookies“, Textdateien, die auf Ihrem Computer gespeichert werden und die eine Analyse der Benutzung der Website durch Sie ermöglichen. Die durch den Cookie erzeugten Informationen über Ihre Benutzung dieser Website werden in der Regel an einen Server von Google in den USA übertragen und dort gespeichert.
					Im Falle der Aktivierung der IP-Anonymisierung auf dieser Webseite, wird Ihre IP-Adresse von Google jedoch innerhalb von Mitgliedstaaten der Europäischen Union oder in anderen Vertragsstaaten des Abkommens über den Europäischen Wirtschaftsraum zuvor gekürzt. Nur in Ausnahmefällen wird die volle IP-Adresse an einen Server von Google in den USA übertragen und dort gekürzt. Die IP-Anonymisierung ist auf dieser Website aktiv. Im Auftrag des Betreibers dieser Website wird Google diese Informationen benutzen, um Ihre Nutzung der Website auszuwerten, um Reports über die Websiteaktivitäten zusammenzustellen und um weitere mit der Websitenutzung und der Internetnutzung verbundene Dienstleistungen gegenüber dem Websitebetreiber zu erbringen.
					Die im Rahmen von Google Analytics von Ihrem Browser übermittelte IP-Adresse wird nicht mit anderen Daten von Google zusammengeführt. Sie können die Speicherung der Cookies durch eine entsprechende Einstellung Ihrer Browser-Software verhindern; wir weisen Sie jedoch darauf hin, dass Sie in diesem Fall gegebenenfalls nicht sämtliche Funktionen dieser Website vollumfänglich werden nutzen können. Sie können darüber hinaus die Erfassung der durch das Cookie erzeugten und auf Ihre Nutzung der Website bezogenen Daten (inkl. Ihrer IP-Adresse) an Google sowie die Verarbeitung dieser Daten durch Google verhindern, indem sie das unter dem folgenden Link verfügbare Browser-Plugin herunterladen und installieren: <a href="http://tools.google.com/dlpage/gaoptout?hl=de">http://tools.google.com/dlpage/gaoptout?hl=de</a>.
					Alternativ zum Browser-Add-On oder innerhalb von Browsern auf mobilen Geräten, <a title="Google Analytics Opt-Out-Cookie setzen" href="javascript:gaOptout()">klicken Sie bitte diesen Link</a>, um die Erfassung durch Google Analytics innerhalb dieser Website zukünftig zu verhindern (das Opt Out funktioniert nur in dem Browser und nur für diese Domain). Dabei wird ein Opt-Out-Cookie auf Ihrem Ger&auml;t abgelegt. L&ouml;schen Sie Ihre Cookies in diesem Browser, m&uuml;ssen Sie diesen Link erneut klicken.

				</p>
			</div>
		</section>

		<section id = "readFile">
			<div id = "fileContainer">
				<div id = "frmUpload">
					<!-- Real Input field, hidden-->
					<input type="file" id="files" name="myFile" onchange="$('#txtFakeText').val($('#files').val().replace('C:\\fakepath\\', '') ); " style="display: none" />
					<input type="text" id="txtFakeText" name="txtFakeText" readonly="true" value="Drop file here or browse">
					<input type="button" id = "fakeBrowse" class="file_btn" onclick="$('#files').click(); " value="BROWSE">
				</div>

				<div id = "fileDrag">Drop file here or browse</div>
				<div id="progress_bar"><div class="percent"></div></div>
				<button id="parse" class="file_btn">LOAD FILE</button>
				<button id="loadTestFile">load a test file</button>
			</div>
		</section>


	</section>

	<section id = "NotSupported">
		<br /> <br />
		<h1>NOT SUPPORTED</h1>
		<p>Your browser lacks required features so it does not support the Blackbird framework! Chrome Browser Recommended!</p>
	</section>

	<br/><br/>
	<section id = "recent">
		<h1>Recent Files</h1>
	</section>

	<br/><br/>
	<section id="about">
		<p class = "descPara">
		<h1>About Blackbird</h1>
			<img src="img/blackbird_logo.png" height="50" style="float:right;padding-left: 6px;">
			Blackbird is a fork of Phinch that implements a couple of new features and takes some other design decisions.
			It contains all features from Phinch version 1.1.2.
			For a detailed list of changes see the <a href="https://github.com/molbiodiv/Blackbird">GitHub repository</a>.
		<h1>About Phinch</h1>
		<img src="img/phinch.png" height="50" style="float:right;padding-left: 6px;">
            Phinch is an open-source framework for visualizing biological data, funded by a grant from the Alfred P. Sloan foundation. This project represents an interdisciplinary collaboration between Pitch Interactive, a data visualization studio in Oakland, CA, and biological researchers at UC Davis. </br></br>
            Whether it's genes, proteins, or microbial species, Phinch provides an interactive visualization tool that allows users to explore and manipulate large biological datasets. Computer algorithms face significant difficulty in identifying simple data patterns; writing algorithms to tease out complex, subtle relationships (the type that exist in biological systems) is almost impossible. However, the human eye is adept at spotting visual patterns, able to quickly notice trends and outliers. It is this philosophy especially when presented with intuitive, well-designed software tools and user interfaces.</br></br>
            
            The sheer volume of data produced from high-throughput sequencing technologies will require fundamentally different approaches and new paradigms for effective data analysis.</br></br>

            Scientific visualization represents an innovative method towards tackling the current bottleneck in bioinformatics; in addition to giving researchers a unique approach for exploring large datasets, it stands to empower biologists with the ability to conduct powerful analyses without requiring a deep level of computational knowledge. </br></br></br>
		</p>
	</section>
	
	<section class = "footer_copyright">
		Blackbird is a fork of Phinch (Copyright © 2014 Phinch, phinch.org).
		This content is released under <a href="http://opensource.org/licenses/BSD-2-Clause" target="_blank">The BSD 2-Clause License</a>.&nbsp;&nbsp;
		<a id="ga-optout" href="javascript:openGaDialog()">This site uses Google Analytics. Click here to learn more or opt-out.</a>
		<span id="ga-optedout" style="display: none">Google Analytics is disabled for you.</span>
		<span id="footer_github">&nbsp;&nbsp;&nbsp;&nbsp;<a href="https://github.com/molbiodiv/Blackbird" target="_blank"><i class="icon-github icon-large"></i>&nbsp;&nbsp;Github</a></span>
	</section>

	<script src="bower_components/biojs-io-biom/build/biom.js"></script>
	<script src="lib/indexedDB.polyfill.js"></script>
	<script src="lib/db.js"></script>
	<script src="lib/jquery-ui-1.10.3.custom.min.js"></script>
	<script src="lib/blob.js"></script>

	<script src="scripts/readFile.js"></script>
	<script src="scripts/init.js"></script>
    <script> var app; window.onload = function() { app = new init('home');} </script>

</body>
</html>
